fecgov/fec-cms
# [DO NOT MERGE]Resolve tech debt related to @extend rule on tables, new CMS table class

## Summary
The  using @extend in CMS-created blocks, which compiles absurdly long CSS selectors and does not accomplish what is was intended to --control tables input by Wagtail editors into html blocks.
This PR does the following;
1) Adds a new class called `.simple-table-cms`, which :
   - will be applied to tables created with the `table block` and `custom-table-block` blocks and all tables created in html blocks in the Wagtail 
   - can also be added to hardcoded tables where applicable and it does not require a class on each table element as the existing `.simple-table` class does.

2) Replaces  `@extend .simple-table` with `@extend .simple-table-cms` in `_richtext.scss` to accomplish the first bullet above.

3) This also solves the issue of absurdly long selectors that were created by `@extend`-ing `.simple_table` class.

- Resolves #2864

## Impacted areas of the application
Statistical press releases where the Wagtail editor (Carmen) uses the custom table block to create the tables
- Any table created using the  `table block` and `custom-table-block` blocks 
- The Home page and  Raising and Spending BarCharts
- The data description tables, for example: http://127.0.0.1:8000/campaign-finance-data/independent-expenditures-file-description/

**Note:** Many of these tables will not be visually changed but we just need to doublecheck to make sure that the addition of `.simple-table-cms ` does not have any unintended effects.

- [ ] - Check with content team to get a list of  tables they may have created in Wagtail using either an  `html  block` or  `table block` or`custom-table-block` so we can test these on the new branch.

## How to test
  - checkout `feature/refactor-extend-rules-add-simple-table-cms`
  - `npm run build-sass`
  - `./manage.py runserver` and check examples of the pages above.
   - http://127.0.0.1:8000/campaign-finance-data/independent-expenditures-file-description/ (Make sure everything looks ok and is not negatively visually altered from what is on production
- http://127.0.0.1:8000/updates/statistical-summary-15-month-campaign-activity-2017-2018-election-cycle/ (Make sure decimal points are still lined up and that the table is not negatively visually altered from what is on production)
- http://127.0.0.1:8000/ (Use Chrome inspector to inspect `simple-table__cell` in the Raising/Spending barcharts and make sure that the really long selectors that are currently on production are reduced to a normal length  )
- Check with content team for tables they may have created in Wagtail using either an  `html  block` or  `table block` or`custom-table-block` to make sure they look fine.
